src/rabbit_stomp_util.erl

```diff
@@ -260,6 +260,33 @@ msg_header_name("1.2") -> ?HEADER_ACK;
 msg_header_name("1.1") -> ?HEADER_MESSAGE_ID;
 msg_header_name("1.0") -> ?HEADER_MESSAGE_ID.
 
+build_arguments(Headers) ->
+    Arguments = lists:foldl(fun({K, V}, Acc) ->
+                                case lists:member(K, ?HEADER_ARGUMENTS) of
+                                  true  -> [build_argument(K, V) | Acc];
+                                  false -> Acc
+                                end
+                            end,
+                            [],
+                            Headers),
+    {arguments, Arguments}.
+
+%% build the actual value thru pattern matching
+build_argument(?HEADER_X_DEAD_LETTER_EXCHANGE, Val) ->
+    {list_to_binary(?HEADER_X_DEAD_LETTER_EXCHANGE), longstr, list_to_binary(string:strip(Val))};
+build_argument(?HEADER_X_DEAD_LETTER_ROUTING_KEY, Val) ->
+    {list_to_binary(?HEADER_X_DEAD_LETTER_ROUTING_KEY), longstr, list_to_binary(string:strip(Val))};
+build_argument(?HEADER_X_EXPIRES, Val) ->
+    {list_to_binary(?HEADER_X_EXPIRES), long, list_to_integer(string:strip(Val))};
+build_argument(?HEADER_X_MAX_LENGTH, Val) ->
+    {list_to_binary(?HEADER_X_MAX_LENGTH), long, list_to_integer(string:strip(Val))};
+build_argument(?HEADER_X_MAX_LENGTH_BYTES, Val) ->
+    {list_to_binary(?HEADER_X_MAX_LENGTH_BYTES), long, list_to_integer(string:strip(Val))};
+build_argument(?HEADER_X_MAX_PRIORITY, Val) ->
+    {list_to_binary(?HEADER_X_MAX_PRIORITY), long, list_to_integer(string:strip(Val))};
+build_argument(?HEADER_X_MESSAGE_TTL, Val) ->
+    {list_to_binary(?HEADER_X_MESSAGE_TTL), long, list_to_integer(string:strip(Val))}.
```

Comment:
it's very declarative and small, though. Without proper macroexpansion capabilities in Erlang I'd leave it as is.